addons/velero/1.6.2/install.sh

```diff
@@ -38,64 +67,100 @@ function velero_join() {
+    "$src"/assets/velero-v"${VELERO_VERSION}"-linux-amd64/velero install --crds-only | grep -v 'Velero is installed'
 
     local resticArg="--use-restic"
     if [ "$VELERO_DISABLE_RESTIC" = "1" ]; then
         resticArg=""
     fi
 
+    # detect if we need to use object store or pvc
     local bslArgs="--no-default-backup-location"
     if ! kubernetes_resource_exists "$VELERO_NAMESPACE" backupstoragelocation default; then
-        bslArgs="--provider aws --bucket $VELERO_LOCAL_BUCKET --backup-location-config region=us-east-1,s3Url=${OBJECT_STORE_CLUSTER_HOST},publicUrl=http://${OBJECT_STORE_CLUSTER_IP},s3ForcePathStyle=true"
+        bslArgs="--provider replicated.com/pvc --bucket velero-internal-snapshots --backup-location-config storageSize=${VELERO_PVC_SIZE},resticRepoPrefix=/var/velero-local-volume-provider/velero-internal-snapshots/restic"
     fi
 
-    velero_credentials
+    # we only need a secret file if it's already set for some other provider (including legacy internal storage)
+    local secretArgs="--no-secret"
+    if kubernetes_resource_exists "$VELERO_NAMESPACE" secret cloud-credentials; then
+        has_secret=true
+        velero_credentials
+        secretArgs="--secret-file velero-credentials"
+    fi
 
-    $src/assets/velero-v${VELERO_VERSION}-linux-amd64/velero install \
+    "$src"/assets/velero-v"${VELERO_VERSION}"-linux-amd64/velero install \
         $resticArg \
         $bslArgs \
-        --plugins velero/velero-plugin-for-aws:v1.2.0,velero/velero-plugin-for-gcp:v1.2.0,velero/velero-plugin-for-microsoft-azure:v1.2.0,replicated/local-volume-provider:v0.1.0,$KURL_UTIL_IMAGE \
-        --secret-file velero-credentials \
-        --use-volume-snapshots=false \
+        $secretArgs \
         --namespace $VELERO_NAMESPACE \
-        --dry-run -o yaml > "$dst/velero.yaml"
+        --plugins velero/velero-plugin-for-aws:v1.2.0,velero/velero-plugin-for-gcp:v1.2.0,velero/velero-plugin-for-microsoft-azure:v1.2.0,replicated/local-volume-provider:v0.3.0,"$KURL_UTIL_IMAGE" \
+        --use-volume-snapshots=false \
+        --dry-run -o yaml > "$dst/velero.yaml" 
 
-    rm velero-credentials
+    if [ -n "$has_secret" ]; then
+        rm velero-credentials
+    fi
 }
 
+# This runs when re-applying the same version to a cluster
 function velero_already_applied() {
     local src="$DIR/addons/velero/$VELERO_VERSION"
     local dst="$DIR/kustomize/velero"
 
-    velero_change_storageclass "$src" "$dst" true
+    determine_velero_pvc_size
 
-    # This should only be applying the configmap if required
-    if compgen -G "$dst/*.yaml" > /dev/null; then
-        kubectl apply -f "$dst"
+    # This function will copy the kustomization template from the file if we need a migration.
+    velero_migrate_from_object_store "$src" "$dst"
+
+    # If we need to migrate, we're going to need to basically reconstruct the original install 
+    # underneath the migration
+    if [ "$WILL_MIGRATE_VELERO_OBJECT_STORE" = "1" ]; then
```

Comment:
```suggestion
    if [ "$BETA_MIGRATE_VELERO_OBJECT_STORE" = "1" ]; then
```

Reply:
Nope, we only need to run this if we are migrating. That variable is checked in `velero_migrate_from_object_store`.